### DISCUSSION TITLE
How to compile libigl on a machine that could not download external dependencies

### DISCUSSION THREAD

**Discussion by hughkhu** [libigl/libigl#2039] [Q&A]
I downloaded the source code of libigl and copied it to a linux machine that **could not connect to the Internet**. When I tried to compile a project with CMake commands like

```
option(LIBIGL_WITH_EMBREE "Use Embree" ON)
include(libigl)
```
, connection timed out error was reported during the download of embree. 

I tried to pre-download embree and put it on the path "libigl/external", but nothing changed and cmake still tried to download embree and rasied timed out error.

How can I solve this problem? Also, I would like to ask if you can put the libigl dependencies in the source code in advance, instead of downloading these dependencies through cmake. Thank you for your help.

**Comment by jdumas** [comment]
If you are on a machine without internet, you should specify [FETCHCONTENT_SOURCE_DIR_<uppercaseName>](https://cmake.org/cmake/help/latest/module/FetchContent.html#variable:FETCHCONTENT_SOURCE_DIR_%3CuppercaseName%3E) for each third-party dependency required by Lagrange, and point the to a locally downloaded folder with the right version of each dependency. You may also want to enable `FETCHCONTENT_FULLY_DISCONNECTED` to prevent CMake from trying to connect to internet.

**Comment by jdumas** [comment]
Locking this discussion in favor of the issue #2040.
